src/utils.rs

```diff
@@ -1,149 +1,90 @@
-use crate::errors::JmbgError;
+use num_enum::TryFromPrimitive;
 
-/// Function to determine gender based on the unique number (BBB)
-pub fn determine_gender(unique_number: u32) -> String {
-    if (0..=499).contains(&unique_number) {
-        "Male".to_string()
-    } else {
-        "Female".to_string()
-    }
+#[derive(Debug, PartialEq, TryFromPrimitive)]
```

Comment:
Does not need TryFromPrimitive, since you are writing your own from unique_number bellow.